### PR TITLE
discord: Helper methods & doc fixes

### PR DESCRIPTION
**Pull request by twoscott**
Just some methods I thought would be convenient. I haven't used most of these structs yet but I figured the CreatedAt methods should be on any object that has a snowflake directly identifying it (not sure if I missed any).

### CODE REVIEW

**Comment by diamondburned** [comment]
Is there a reason why `CreatedAt` returns `discord.Timestamp` and not `time.Time`?

**Comment by twoscott** [comment]
I honestly just wasn't sure which to use, figured you might prefer to use `discord.Timestamp` to keep consistency but I could see `time.Time` working just as well if not better

**Comment by diamondburned** [comment]
`discord.Timestamp` would indeed be more consistent, but since these are methods, it doesn't really matter. The type was made just so JSON can be marshaled properly, so it doesn't matter as far as usage goes.

File: discord/emoji.go [hunk 1]
```diff
@@ -35,12 +36,29 @@ type Emoji struct {
...
 }
 
+// IsCustom returns whether the emoji is a custom emoji.
+func (e Emoji) IsCustom() bool {
+	return !e.IsUnicode()
+}
+
+// IsUnicode returns whether the emoji is a unicode emoji.
+func (e Emoji) IsUnicode() bool {
+	return e.ID.IsNull()
```
Comment (diamondburned) [inline discord/emoji.go]: It would be better to use `.IsValid()` here.

**Review by diamondburned — CHANGES_REQUESTED** [review]

**Review by diamondburned — APPROVED** [review]

**Pull request merged**